internal/pkg/server/fleet.go

```diff
@@ -193,8 +194,54 @@ LOOP:
 		curCfg = newCfg
 
 		select {
-		case newCfg = <-f.cfgCh:
+		case cfg := <-f.cfgCh:
 			log.Info().Msg("Server configuration update")
+			if cfg.Inputs == nil && cfg.RevisionIdx != 0 { // cfg only contains updated output retrieved from policy
+				rev := cfg.RevisionIdx
+				esOutput := config.MergeElasticsearchFromPolicy(curCfg.Output.Elasticsearch, cfg.Output.Elasticsearch)
+
+				// test config
+				cli, err := es.NewClient(ctx,
+					&config.Config{
+						Output: config.Output{
+							Elasticsearch: esOutput,
+						},
+					},
+					false,
+					elasticsearchOptions(curCfg.Inputs[0].Server.Instrumentation.Enabled, f.bi)...,
+				)
+				if err != nil {
+					log.Warn().Int64(logger.RevisionIdx, rev).Err(err).Msg("unable to create elasticsearch client from policy output")
+					continue
+				}
+				remoteVersion, err := ver.CheckCompatibility(ctx, cli, f.bi.Version)
+				if err != nil {
+					// NOTE The error can indicate a bad network connection, bad TLS settings, etc.
+					// But if the error is an ErrElasticVersionConflict then something is very wrong
+					if errors.Is(err, es.ErrElasticVersionConflict) {
+						log.Error().Err(err).Int64(logger.RevisionIdx, rev).Interface("output", esOutput).Interface("bootstrap", curCfg.Output.Elasticsearch).Str("remote_version", remoteVersion).Msg("Elasticsearch version constraint failed for new output")
+					} else {
+						log.Warn().Err(err).Int64(logger.RevisionIdx, rev).Msg("Failed version compatibility check using output from policy")
+					}
+					continue
+				}
+				// work around to get a new cfg object based off curCfg
+				// we override the output with esOutput and have a complete config with a new mutex
+				tmp, err := ucfg.NewFrom(curCfg, config.DefaultOptions...)
+				if err != nil {
+					log.Error().Err(err).Int64(logger.RevisionIdx, rev).Msg("Unable to convert config")
+					continue
+				}
+				err = tmp.Unpack(cfg, config.DefaultOptions...)
+				if err != nil {
+					log.Error().Err(err).Int64(logger.RevisionIdx, rev).Msg("Unable to unpack config")
```

Comment:
When running under a real agent unpack results in an error: `verification mode must be an identifier accessing 'inputs.0.server.ssl.verification_mode' accessing 'inputs.0.server.ssl'`
I'll try to do something else to resolve this; but it might need a change to elastic-agent-libs

Reply:
pr: https://github.com/elastic/elastic-agent-libs/pull/196

Reply:
another pr: https://github.com/elastic/elastic-agent-libs/pull/197

I made a custom build of the fleet-server and ran it with a real agent under a real cluster and found we now see the successful "Using output from policy" log message, but do not see the correct output in the diagnostics bundle. I'm going to see if there is a straightforward way to pass the config to a method in `agent.go` instead of `fleet.go` so we can get functional diagnostics